client/my-sites/controller.js

```diff
@@ -560,14 +561,14 @@ export function siteSelection( context, next ) {
 
 	/*
 	 * If the user has only one site, redirect to the single site context instead of
-	 * rendering the all-site views.
+	 * rendering the all-site views. Exclude plugins page from this behavior.
 	 *
 	 * If the primary site is not yet available in Redux state, initiate a fetch and postpone the
 	 * redirect until the fetch is complete. (while the primary site ID is a property of the
 	 * current user object and therefore always available, we need to fetch the site info in order
 	 * to convert the site ID to the site slug that will be part of the redirect URL)
 	 */
-	if ( hasOneSite && ! siteFragment ) {
+	if ( hasOneSite && ! siteFragment && ! isNextPathPlugins ) {
```

Comment:
Maybe we could change the implementation to a blacklist of paths so that the solution scales if we decide that we need to blacklist more/new paths, what do you think?

Reply:
Makes sense! I've addressed it on https://github.com/Automattic/wp-calypso/pull/94330/commits/9373fc060e65a7420d19a83af20c619907d955a7